app/src/main/java/org/schabi/newpipe/download/DownloadDialog.java

```diff
@@ -562,6 +562,24 @@ public void onItemSelected(final AdapterView<?> parent,
                 selectedSubtitleIndex = position;
                 break;
         }
+        onItemSelectedSetFileName();
+    }
+
+    private void onItemSelectedSetFileName() {
+        final String setSubtitleLanguageCode = subtitleStreamsAdapter.getItem(selectedSubtitleIndex)
+                .getLanguageTag();
```

Comment:
Move this in between lines 578 and 579 (add `{}` around the `switch` branch if the compiler complains).

Reply:
Done